pumps/graph_sql_test.go

```diff
@@ -0,0 +1,290 @@
+				if err := pump.db.Migrator().DropTable(conf.TableName); err != nil {
+					t.Error(err)
+				}
+			})
+			records := make([]interface{}, 0)
+			expectedResponses := make([]analytics.GraphRecord, 0)
+			// create the records to passed to the pump
+			for _, item := range tc.records {
+				r := analytics.AnalyticsRecord{
+					APIName: "Test API",
+					Path:    "POST",
+					Tags:    item.tags,
+				}
+				if !item.isHTTP {
+					r.RawRequest = convToBase64(rawGQLRequest)
+					r.ApiSchema = convToBase64(schema)
+				} else {
+					r.RawRequest = convToBase64(rawHTTPReq)
+					r.RawResponse = convToBase64(rawHTTPResponse)
+				}
+				r.RawResponse = convToBase64(item.response)
+				if item.responseCode != 0 {
+					r.ResponseCode = item.responseCode
+				}
+				records = append(records, r)
+			}
+
+			// create the responses to be expected from the db
+			for _, item := range tc.responses {
+				r := analytics.GraphRecord{
+					Types:         item.types,
+					OperationType: item.operationType,
+					Errors:        []analytics.GraphError{},
+				}
+				if item.expectedErr == nil {
+					r.Errors = []analytics.GraphError{}
+				} else {
+					r.Errors = item.expectedErr
+					r.HasErrors = true
+				}
+				expectedResponses = append(expectedResponses, r)
+			}
+
+			err := pump.WriteData(context.Background(), records)
+			if !tc.hasError {
+				r.NoError(err)
+			} else {
+				r.Error(err)
+			}
+
+			var resultRecords []analytics.GraphRecord
+			pump.db.Table(conf.TableName).Find(&resultRecords)
+			r.Equalf(len(tc.responses), len(resultRecords), "responses count do no match")
+			if diff := cmp.Diff(expectedResponses, resultRecords, cmpopts.IgnoreFields(analytics.GraphRecord{}, "AnalyticsRecord")); diff != "" {
+				t.Error(diff)
+			}
+		})
+	}
+}
+
+func TestGraphSQLPump_Sharded(t *testing.T) {
+	r := require.New(t)
+	conf := SQLConf{
+		Type:             "sqlite",
+		ConnectionString: "",
+		TableName:        "graph-record",
+		TableSharding:    true,
+	}
+	pump := &GraphSQLPump{}
+	assert.NoError(t, pump.Init(conf))
+
+	baseRecord := analytics.AnalyticsRecord{
+		APIID:        "test-api",
+		Path:         "/test-api",
+		RawRequest:   convToBase64(rawGQLRequest),
+		RawResponse:  convToBase64(rawGQLResponse),
+		ApiSchema:    convToBase64(schema),
+		Tags:         []string{analytics.PredefinedTagGraphAnalytics},
+		APIName:      "test-api",
+		ResponseCode: 200,
+		Method:       "POST",
+	}
+
+	expectedTables := make([]string, 0)
+	records := make([]interface{}, 0)
+	for i := 1; i <= 3; i++ {
+		day := i
+		timestamp := time.Date(2023, time.January, day, 0, 1, 0, 0, time.UTC)
+		rec := baseRecord
+		rec.TimeStamp = timestamp
+		rec.Month = timestamp.Month()
+		rec.Day = timestamp.Day()
+		rec.Year = timestamp.Year()
+		records = append(records, rec)
+		expectedTables = append(expectedTables, fmt.Sprintf("%s_%s", conf.TableName, timestamp.Format("20060102")))
+	}
+
+	r.NoError(pump.WriteData(context.Background(), records))
```

Comment:
should we clean up the table after this?

Reply:
Added cleanup in L305